### PR TITLE
Give more clear error message when gscope is non-empty

### PR DESCRIPTION
**Pull request by JackCaoG**
Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by JackCaoG** [comment]
I have signed the CLA, not sure when this will be updated on this page.

**Review by alanwaketan — APPROVED** [review]
LGTM!

**Comment by alanwaketan** [comment]
Just repeat my question from the xla PR here: any change we can have that test case in the upstream too?

File: torch/csrc/lazy/core/ir_metadata.cpp [hunk 1]
```diff
@@ -61,7 +73,10 @@ void PopScope() {
 }
 
 void ResetScopeContext() {
-  TORCH_CHECK(g_scope_context.scopes.size() == 0);
+  if (g_scope_context.scopes.size() != 0) {
```
Comment (Krovatkin) [inline torch/csrc/lazy/core/ir_metadata.cpp:76]: this is fine, but we could probably just use `TORCH_CHECK`?

`TORCH_CHECK(g_scope_context.scopes.size() == 0, "Expecting scope to be empty but it is " + GetCurrentScope())`

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Comment by Krovatkin** [comment]
@JackCaoG looks like you guys need to sign the new Linux CLA?

**Pull request closed without merging**